From 9c79cd25e2b2c81860484e0ba3993e2e42964847 Mon Sep 17 00:00:00 2001
From: yzhu1 <yanjun.zhu@windriver.com>
Date: Sun, 11 May 2014 22:45:25 -0700
Subject: [PATCH 1/2] icmp: account for ICMP out errors because of socket limit

This patch is similar to commit b32c6578 [icmp: Account for ICMP out errors].
When icmp_xmit_lock fails because of socket limit or memory shortage,
increment ICMP_MIB_OUTERRORS counter, so that "netstat -s" can report
these errors.

netstat -s | grep "ICMP messages failed"
    0 ICMP messages failed

Signed-off-by: yzhu1 <yanjun.zhu@windriver.com>
---
 net/ipv4/icmp.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/net/ipv4/icmp.c b/net/ipv4/icmp.c
index 4a5137a..4b29b2e 100644
--- a/net/ipv4/icmp.c
+++ b/net/ipv4/icmp.c
@@ -366,8 +366,10 @@ static void icmp_reply(struct icmp_bxm *icmp_param, struct sk_buff *skb)
 		return;
 
 	sk = icmp_xmit_lock(net);
-	if (sk == NULL)
+	if (sk == NULL){
+		ICMP_INC_STATS_BH(net, ICMP_MIB_OUTERRORS);
 		return;
+	}
 	inet = inet_sk(sk);
 
 	icmp_param->data.icmph.checksum = 0;
-- 
1.7.0

